pkg/ipnet/ipnet.go

```diff
@@ -3,8 +3,11 @@ package ipnet
 
 import (
 	"encoding/json"
+	"fmt"
 	"net"
 	"reflect"
+
+	yaml "gopkg.in/yaml.v2"
```

Comment:
We should  be using `github.com/ghodss/yaml`

Kubernetes only does JSON marshal/unmarshal; `github.com/ghodss/yaml` does the jsontoyaml.
No need for special YAML parsing.

Reply:
> Kubernetes only does JSON marshal/unmarshal; github.com/ghodss/yaml does the jsontoyaml.

Adding explicit YAML support to `pkg/ipnet` was only 40 lines of non-test additions, so it seemed reasonable to me.  Are we sure this is only ever going to be consumed internally?

The immediate motivation was that `installer/pkg/config` currently [uses github.com/ghodss/yaml][1], presumably to support legacy properties with different YAML and JSON serialization (like [`BaseDomain`]2]).  If we wanted to only support github.com/ghodss/yaml with ipnet and `pkg/types`, then we'd need to import both packages in `installer/pkg/config` with distinct names and use the right name depending on whether we wanted the nominally-JSON or nominally-YAML tags for a given (un)marshal.  It seemed  less confusing to just add support for the library installer/pkg/config was already using.

[1]: https://github.com/openshift/installer/blob/84ab99e316a3f4f768b2ff145b8094c2779597c1/installer/pkg/config/cluster.go#L8
[2]: https://github.com/openshift/installer/blob/84ab99e316a3f4f768b2ff145b8094c2779597c1/installer/pkg/config/cluster.go#L76

Reply:
https://github.com/openshift/installer/blob/84ab99e316a3f4f768b2ff145b8094c2779597c1/installer/pkg/config/cluster.go#L76 this was done because they didn't want to write code to create terraform.tfvars.

Reply:
> this was done because they didn't want to write code to create `terraform.tfvars`.

So if I drop our direct `gopkg.in/yaml.v2` use, do you want me to write separate code to create `terraform.tfvars`?

Reply:
`terraform.tfvars` is definitely a separate structure.
we shouldn't have done what we did with `installer.Config` to make is directly marshal into `terraform.tfvars`. 
The `installer.Config` ended up with fields like `IgnitionMasters` that have `yaml:"-", as we didn't want user's to specify.